src/driver/express/ExpressDriver.ts

```diff
@@ -117,15 +118,19 @@ export class ExpressDriver extends BaseDriver implements Driver {
                 if (isPromiseLike(checkResult)) {
                     checkResult.then(result => {
                         if (!result) {
-                            return this.handleError(new AccessDeniedError(action), actionMetadata, action);
+                            return this.handleError((actionMetadata.authorizedRoles.length === 0
+                                    ? new AuthorizationRequiredError(action) : new AccessDeniedError(action)),
+                                actionMetadata, action);
 
                         } else {
                             next();
                         }
                     });
                 } else {
                     if (!checkResult) {
-                        this.handleError(new AccessDeniedError(action), actionMetadata, action);
+                        this.handleError((actionMetadata.authorizedRoles.length === 0
```

Comment:
@19majkel94 you should investigate a bit how `Promise.resolve` actually works, its not as simple as let vs const. We make decisions based on pros and cons of everything, its impossible to be consistent in everything, so make judging based on each scenario individually.